flink-core/src/test/java/org/apache/flink/api/common/io/GlobFilePathFilterTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.io;
+
+import org.apache.flink.core.fs.Path;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class GlobFilePathFilterTest {
+	@Test
+	public void defaultConstructorCreateMatchAllFilter() {
+		GlobFilePathFilter matcher = new GlobFilePathFilter();
+		assertFalse(matcher.filterPath(new Path("dir/file.txt")));
+	}
+
+	@Test
+	public void matchAllFilesByDefault() {
+		GlobFilePathFilter matcher = new GlobFilePathFilter(
+			Collections.<String>emptyList(),
+			Collections.<String>emptyList());
+
+		assertFalse(matcher.filterPath(new Path("dir/file.txt")));
+	}
+
+	@Test
+	public void excludeFilesNotInIncludePatterns() {
+		GlobFilePathFilter matcher = new GlobFilePathFilter(
+			Collections.singletonList("dir/*"),
+			Collections.<String>emptyList());
+
+		assertFalse(matcher.filterPath(new Path("dir/file.txt")));
+		assertTrue(matcher.filterPath(new Path("dir1/file.txt")));
+	}
+
+	@Test
+	public void excludeFilesIfMatchesExclude() {
+		GlobFilePathFilter matcher = new GlobFilePathFilter(
+			Collections.singletonList("dir/*"),
```

Comment:
Is it possible to match file names with star in it, e.g. by escaping `\*` to literally match `*`?